google/cloud/storage/internal/generate_message_boundary_benchmark.cc

```diff
@@ -0,0 +1,156 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/internal/generate_message_boundary.h"
+#include "google/cloud/internal/random.h"
+#include <benchmark/benchmark.h>
+#include <mutex>
+#include <random>
+#include <string>
+
+namespace google {
+namespace cloud {
+namespace storage {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace internal {
+namespace {
+
+// CXX=clang++ CC=clang bazel run -c opt --
+//   //google/cloud/storage:internal_generate_message_boundary_benchmark
+//   --benchmark_repetitions=100
+//
+// clang-format off
+// Run on (96 X 2000 MHz CPU s)
+// CPU Caches:
+//  L1 Data 32 KiB (x48)
+//  L1 Instruction 32 KiB (x48)
+//  L2 Unified 1024 KiB (x48)
+//  L3 Unified 39424 KiB (x2)
+// ---------------------------------------------------------------------------------------------
+// Benchmark                                                   Time             CPU   Iterations
```

Comment:
Opinion: I usually think that the minimum is the best value to record (instead of mean or median) for such micro-benchmarks.